pkg/asset/tls/stock.go

```diff
@@ -0,0 +1,297 @@
+package tls
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+// Stock is the stock of TLS assets that can be generated.
+type Stock interface {
+	// RootCA is the asset that generates the root-ca key/cert pair.
+	RootCA() asset.Asset
+	// KubeCA is the asset that generates the kube-ca key/cert pair.
+	KubeCA() asset.Asset
+	// EtcdCA is the asset that generates the etcd-ca key/cert pair.
+	EtcdCA() asset.Asset
+	// AggregatorCA is the asset that generates the aggregator-ca key/cert pair.
+	AggregatorCA() asset.Asset
+	// ServiceServingCA is the asset that generates the service-serving-ca key/cert pair.
+	ServiceServingCA() asset.Asset
+	// EtcdClientCertKey is the asset that generates the etcd client key/cert pair.
+	EtcdClientCertKey() asset.Asset
+	// AdminCertKey is the asset that generates the admin key/cert pair.
+	AdminCertKey() asset.Asset
+	// IngressCertKey is the asset that generates the ingress key/cert pair.
+	IngressCertKey() asset.Asset
+	// APIServerCertKey is the asset that generates the API server key/cert pair.
+	APIServerCertKey() asset.Asset
+	// OpenshiftAPIServerCertKey is the asset that generates the Openshift API server key/cert pair.
+	OpenshiftAPIServerCertKey() asset.Asset
+	// APIServerProxyCertKey is the asset that generates the API server proxy key/cert pair.
+	APIServerProxyCertKey() asset.Asset
+	// KubeletCertKey is the asset that generates the kubelet key/cert pair.
+	KubeletCertKey() asset.Asset
+	// TNCCertKey is the asset that generates the TNC key/cert pair.
+	TNCCertKey() asset.Asset
+	// ClusterAPIServerCertKey is the asset that generates the cluster API server key/cert pair.
+	ClusterAPIServerCertKey() asset.Asset
+	// ServiceAccountKeyPair is the asset that generates the service-account key/cert pair.
```

Comment:
nit: "key/cert" -> "key".

Reply:
It is generating both public and private key pair, why just saying "key"?

Reply:
> It is generating both public and private key pair, why just saying "key"?

Making a "key/cert" -> "key" replacement will leave us with "...generates the service-account key pair.".  I'm fine if you want to call out public/private as well, but I don't want to claim there's a cert involved.

Reply:
updated